nvse/nvse/ArrayVar.h

```diff
@@ -82,54 +93,67 @@ struct ArrayData
 	};
 
 	~ArrayData();
-	const char *GetStr() const;
+	ArrayData() = default;
+	[[nodiscard]] const char *GetStr() const;
 	void SetStr(const char *srcStr);
 
+	//Casts the data in the form InternalFunctionCaller::PopulateArgs() understands.
+	[[nodiscard]] void* GetAsVoidArg() const;
+
 	ArrayData& operator=(const ArrayData &rhs);
+	ArrayData(const ArrayData &from);
 };
 STATIC_ASSERT(sizeof(ArrayData) == 0x10);
 
-struct ArrayElement
+template <bool isSelfOwning = false>
+struct ArrayElement_Templ
```

Comment:
I tried to use inheritance first but got scared away as I realized the base parent's destructor would also be called. 
It might be fine, might have to use a `virtual` destructor or something, but apparently those resolve calls to functions at compile-time, which sounds bad.

Plus I would have to entirely copy ArrayElement's `Unset()` to do a minor tweak, which isn't great in case it needs to be refactored (would have to change it twice, since the derived class adds another instance).

But yeah it is very intrusive. So I can change it if you're sure about the order of destructor calls with a derived class.